products/sle/main.pm

```diff
@@ -418,6 +419,47 @@ logcurrentenv(
       SLE_PRODUCT SPLITUSR VIDEOMODE)
 );
 
+sub parse_condition {
+    my ($schedule, $condition_name) = @_;
+    # Get condition hash
+    my $condition = $schedule->{conditional_schedule}->{$condition_name};
+    # Iterate over variables in the condition
+    foreach my $var (keys %{$condition}) {
+        next unless my $val = get_var($var);
+        # If value of the variable matched the conditions
+        # Iterate over the list of the modules to be loaded
+        for my $test (@{$condition->{$var}->{$val}}) {
+            loadtest $test;
+        }
+    }
+}
+
+sub parse_test_suite_setting {
+    my ($setting) = shift;
+    # allow comma spearated list
+    for my $test (split(',', get_var($setting))) {
```

Comment:
Take into account that there could be tests that didn't have the settings properly processed and end up being %XYZ%, it's a border scenario, but you know :)